deps/captnlog/src/captnlog.c

```diff
@@ -214,3 +214,29 @@ void captainslog_deinit()
 #endif
     }
 }
+
+char *captainslog_alloc_varg_str(int level, const char *fmt, ...)
+{
+    char *str;
+
+    if (level > g_state.level || !g_state.initialized) {
+        return NULL;
+    }
+
+    str = malloc(512);
+
+    if (str != NULL) {
+        va_list args;
+        va_start(args, fmt);
+        vsnprintf(str, 512, fmt, args);
+        va_end(args);
+    }
+    return str;
+}
+
+void captainslog_free_varg_str(char *str)
+{
+    if (str != NULL) {
+        free(str);
```

Comment:
Standard says free should be fine to call on a NULL pointer so the null check is redundant.

Reply:
According to this stackoverflow there are some library implementations where it will incur crash. See
https://stackoverflow.com/questions/1938735/does-freeptr-where-ptr-is-null-corrupt-memory

But I can remove check, no problem.

Reply:
Fixed.